src/schoology-extractor/schoology_extractor/helpers/sync.py

```diff
@@ -169,11 +177,20 @@ def sync_resource(
 
     for row in data:
         hasChanged = _resource_has_changed(row, resource_name, db_engine, id_column) if table_exist else True
-        row["CreateDate"] = _get_created_date(resource_name, db_engine, id_column, row[id_column]) if table_exist else current_date_with_format
-        row["LastModifiedDate"] = current_date_with_format if hasChanged else _get_last_modified_date(resource_name, db_engine, id_column, row[id_column])
+        create_date = _get_created_date(resource_name, db_engine, row[id_column]) if table_exist else current_date_with_format
+        last_modified_date = current_date_with_format if hasChanged else _get_last_modified_date(resource_name, db_engine, row[id_column])
+        record = dict(
+            CreateDate=create_date,
+            LastModifiedDate=last_modified_date,
+            SourceId=row[id_column],
+            JsonResponse=json.dumps(row)
+        )
+        data_to_write.append(record)
+        row["CreateDate"] = create_date
+        row["LastModifiedDate"] = last_modified_date
 
-    df_data = DataFrame(data)
-    _write_resource_to_db(resource_name, db_engine, df_data, column_mapping)
-    _remove_duplicates(resource_name, db_engine, id_column)
+    df_data = DataFrame(data_to_write)
```

Comment:
2 Different formats are needed:
1. Used for syncDB
```python3 
record = dict(
            CreateDate=create_date,
            LastModifiedDate=last_modified_date,
            SourceId=row[id_column],
            JsonResponse=json.dumps(row)
        )
```

2. Used for mapping and exporting to csv.
```python3
# row is the original item, we add this properties to it with the help of the sync process
row["CreateDate"] = create_date
row["LastModifiedDate"] = last_modified_date
```

For that reason, we need two dataframes: one as output of the process, another one to write to sync_db.